pkgs/package-overrides.nix

```diff
@@ -44,6 +44,15 @@ in
       else
         prev.extensions.apcu;
 
+    couchbase = prev.extensions.couchbase.overrideAttrs (attrs: {
+      internalDeps =
+        attrs.internalDeps or []
+        ++ lib.optionals (lib.versionOlder prev.php.version "8.0") [
+          #TODO: How to reference the json extension from this file in here?
+          prev.extensions.json
```

Comment:
The JSON extension is defined in this layer, not the previous one. So you need to get it from the `final` package set.